lib/rubocop/cop/performance/string_replacement.rb

```diff
@@ -20,10 +20,15 @@ module Performance
       #   'a b c'.delete(' ')
       class StringReplacement < Cop
         MSG = 'Use `%s` instead of `%s`.'
-        DETERMINISTIC_REGEX = /^[\w\s\-,."']+$/
-        REGEXP_CONSTRUCTOR_METHODS = [:new, :compile]
-        GSUB_METHODS = [:gsub, :gsub!]
-        DETERMINISTIC_TYPES = [:regexp, :str, :send]
+        DETERMINISTIC_REGEX = /^[\w\s\-,."']+$/.freeze
+        REGEXP_CONSTRUCTOR_METHODS = [:new, :compile].freeze
+        GSUB_METHODS = [:gsub, :gsub!].freeze
+        DETERMINISTIC_TYPES = [:regexp, :str, :send].freeze
+        DELETE = 'delete'.freeze
+        TR = 'tr'.freeze
+        BANG = '!'.freeze
+        SINGLE_QUOTE = "'".freeze
```

Comment:
I had the same thought as I was working on some performance/memory tweaks to the app. It is a slow process to convert everything. It could be broken up into small chunks though.